pom.xml

```diff
@@ -546,6 +546,7 @@
               -Xep:OperatorPrecedence:OFF \
               -Xep:ReferenceEquality:OFF \
               -Xep:UnnecessaryParentheses:OFF \
+              -Xep:InlineMeSuggester:OFF \
```

Comment:
I am disabling here the `InlineMeSuggester` which causes the build to fail if the annotation target can be inlined; in this case

https://errorprone.info/bugpattern/InlineMeSuggester. 

```
[WARNING] /Users/evacchi/Devel/java/chicory/runtime/src/main/java/com/dylibso/chicory/runtime/HostFunction.java:[15,12] [InlineMeSuggester] This deprecated API looks inlineable. If you'd like the body of the API to be automatically inlined to its callers, please annotate it with @InlineMe. NOTE: the suggested fix makes the method final if it was not already.
    (see https://errorprone.info/bugpattern/InlineMeSuggester)
  Did you mean '@InlineMe(replacement = "this(moduleName, fieldName, handle, paramTypes, returnTypes)")'?
```

while this is useful, it requires to add a run-time dependency on error-prone, which I assume we want to avoid.

Reply:
> it requires adding a run-time dependency on error-prone

this should not be accurate, error-prone is used only in this repository and is not required as a dependency as it's declared in the `annotationProcessorPaths` section.

Instead of fully disabling the check I suggest (if applicable):
- a local warning suppression
- to fix the warning

[example reference](https://github.com/dylibso/chicory/blob/9594ed738d7a64976d47def85e0f917c2644c339/log/src/main/java/com/dylibso/chicory/log/Logger.java#L78) cc. @electrum

Reply:
no, I mean that adding `@InlineMe` as suggested requires adding the run-time dependency to the artifact that contains that annotation. 

I can indeed add a local SuppressWarning as I originally did in this commit https://github.com/dylibso/chicory/pull/527/commits/0d5e01d713d68cd56290a8ecd7013e96bfc6a569. I can revert the global disable if you prefer. 

TBH I can also remove the `@Deprecated` method altogether; I felt it was useful to give a short grace period to end users; but OTOH we are indeed renaming quite a few other symbols...

Reply:
> I can also remove the `@Deprecated` method altogether

Strongly agree, we don't want to have `@Deprecated` code in a 1.0 and, since the last release, we are already beyond any possible "grace period" for the user.
At the moment, we are releasing `0.0.x` versions, and API breaking changes are expected.